From 03d7fd6c6b21150475700752b90d10c1255b11ff Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Tue, 11 Feb 2025 22:10:22 -0800
Subject: [PATCH] Use idomatic approach to extending CMAKE_MODULE_PATH.

There are many reasons why there may already be a CMAKE_MODULE_PATH defined. The idiomatic way to extend it in a project is via `list(PREPEND`.

Co-authored-by: Marius Brehler <marius.brehler@amd.com>
---
 CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index df6755a..f6d3a7e 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,10 +49,8 @@ endif()
 # Generate static package, when BUILD_SHARED_LIBS is set to OFF.
 # Default to ON
 option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
-## Set default module path if not already set
-if(NOT DEFINED CMAKE_MODULE_PATH)
-    set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/")
-endif()
+
+list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules")
 ## Include common cmake modules
 include(utils)
 
-- 
2.43.0

